app/assets/javascripts/repositories/repository_datatable.js.erb

```diff
@@ -1082,7 +1085,19 @@ var RepositoryDatatable = (function(global) {
   }
 
   global.clearFileInput = function (el) {
-    $(el).prev('div').find('input:file')[0].value = '';
+    var parent = $(el).closest('div.repository-input-file-field');
+    var input  = parent.find('input:file')[0];
+    // hide clear button
+    $(parent.find('a[data-action="removeAsset"]')[0]).hide();
+    // reset value
+    input.value = '';
+    // add flag
+    $(input).attr('remove', true);
+    // clear fileName
+    $(parent.find('.file-name-label')[0]).text('');
```

Comment:
Maybe, instead of `''`, here we should also display `'No file chosen'` text (localized if possible).